packages/vuetify/src/components/VFileInput/VFileInput.ts

```diff
@@ -0,0 +1,213 @@
+// Styles
+import './VFileInput.sass'
+
+// Extensions
+import VTextField from '../VTextField'
+
+// Components
+import { VChip } from '../VChip'
+
+// Types
+import { PropValidator } from 'vue/types/options'
+
+// Utilities
+import { humanReadableFileSize, wrapInArray } from '../../util/helpers'
+
+export default VTextField.extend({
+  name: 'v-file-input',
+
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+
+  props: {
+    accept: {
+      type: String,
+      default: null,
+    } as PropValidator<string | null>,
+    chips: Boolean,
+    clearable: {
+      type: Boolean,
+      default: true,
+    },
+    counterSizeString: {
+      type: String,
+      default: '$vuetify.fileInput.counterSize',
+    },
+    counterString: {
+      type: String,
+      default: '$vuetify.fileInput.counter',
+    },
+    displaySize: {
+      type: [Boolean, Number],
+      default: false,
+      validator: (v: boolean | number) => {
+        return (
+          typeof v === 'boolean' ||
+          [1000, 1024].includes(v)
+        )
+      },
+    } as PropValidator<boolean | 1000 | 1024>,
+    multiple: Boolean,
+    placeholder: String,
+    prependIcon: {
+      type: String,
+      default: '$vuetify.icons.file',
+    },
+    readonly: {
+      type: Boolean,
+      default: true,
+    },
+    smallChips: Boolean,
+    type: {
+      type: String,
+      default: 'file',
+    },
+    value: {
+      type: [Object, Array],
+      default: () => [],
+    } as PropValidator<File | File[]>,
+  },
+
+  data () {
+    return {
+      lazyValue: wrapInArray(this.value) as File[],
+    }
+  },
+
+  computed: {
+    classes (): object {
+      return {
+        ...VTextField.options.computed.classes.call(this),
+        'v-file-input': true,
+      }
+    },
+    counterValue (): string {
+      if (!this.displaySize) return this.$vuetify.lang.t(this.counterString, this.lazyValue.length)
+
+      const bytes = this.internalValue.length > 0
+        ? (this.internalValue as File[])
+          .map(f => f.size)
+          .reduce((a, b) => a + b)
+        : 0
+
+      return this.$vuetify.lang.t(
+        this.counterSizeString,
+        this.lazyValue.length,
+        humanReadableFileSize(bytes, this.base)
+      )
+    },
+    internalValue: {
+      get (): File[] {
+        return this.lazyValue
+      },
+      set (val: File | File[]) {
+        this.lazyValue = wrapInArray(val)
+        this.$emit('change', this.lazyValue)
+      },
+    },
+    isDirty (): boolean {
+      return this.internalValue.length > 0
```

Comment:
```suggestion
      return this.internalValue.length
```

Reply:
`this.internalValue.length` is not `boolean`